### ISSUE TITLE
Add support for `.env.local` files

### ISSUE THREAD

**Issue by markostimac** [drwatson1/dbup-cli#7]
The variables listed in the `.env` file can be used by different tools/applications - the application code can read a connection string from `.env` file as well as `dbup-cli` tool.

Since `.env` files are usually source controlled it is common practice to provide overrides for default values specified in `.env.local` file.

Would it be possible to add support for `.env.local` files to `dbup-cli`?

I guess the most naive solution could be to just extend the `LoadEnvironmentVariables` method and check if `.env.local` exists and load the variables.

**Comment by drwatson1** [comment]
Thank you for the suggestion!

I think it is possible, of course. Let's clear it up.

Now, environment variables are loaded in the following order:

1. OS (process) environment variables;
1. '.env' in the current folder;
1. '.env' next to a configuration file, whether default 'dbup.yml' file or specified via command line;
1. Additional files, specified in the command line.

I think `.env.local` should be loaded after n.3 and before n.4. Am I right? Did you mean exactly that?
If so, would it be enough to load `.env.local` placed next to the `dbup.yml` or it would be better to load the file from the current folder before that?

I'm looking forward to your opinion.

By the way, you can use the fourth option to solve your problem for now:

```bash
dbup upgrade -e .env.local
```

**Comment by markostimac** [comment]
Thank you for answering so quickly!

I'm always running `dbup-cli` from the root of my project because this is where I keep the `dbup.yml` file.

Let's say this is the directory structure:
```
root
 |- src
    |- migrations
       |- 001.sql
       |- .env
       |- .env.local
 |- .env
 |- .env.local
 |- additional-vars.env
 |- dbup.yml 
```
If I apply migrations like this: `dotnet dbup upgrade -e additional-vars.env`, I would expect variables to be read in the following order:
 1. OS (process) environment variables
 2. root/.env
 3. root/.env.local
 4. root/src/migrations/.env
 5. root/src/migrations/.env.local
 6. root/additional-vars.env

Does this make sense? Please let me know if this fits the current architecture (it's possible I misunderstood something :))

**Comment by drwatson1** [comment]
Subfolders are not supported for searching `.env` files. As I wrote [here](https://github.com/drwatson1/dbup-cli/wiki#using-env-files), the file from a current folder will be loaded first, the file next to `dbup.yml` will be loaded after that. No subfolders will be searched, so the `src/migrations/.env` will not be loaded in any case.

The only way to use the `src/migrations/.env` file is to run the `dbup` from the `src/migrations` folder:

```bash
cd src/migrations
dbup upgrade ../../dbup.yml
```

The order will be:
- `src/migrations/.env`
- `.env`

When I implement loading a `.env.local`, the order will be:
- `src/migrations/.env`
- `src/migrations/.env.local`
- `.env`
- `.env.local`

Is that suitable for you?

**Comment by drwatson1** [comment]
Implementation was pretty simple. I attached an updated version. Can you try it? Is this enough?

[dbup-cli.zip](https://github.com/drwatson1/dbup-cli/files/6177665/dbup-cli.zip)

**Comment by markostimac** [comment]
Thanks, @drwatson1, that sounds perfect. I'll try it out as soon as I make some time. Thanks again!

**Comment by drwatson1** [comment]
I released a new version [v1.4.0](https://github.com/drwatson1/dbup-cli/releases/tag/v1.4.0) with a requested feature. Feel free to reopen the issue in case of any problems or create a new issue.
